network/tlskc.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"math/big"
+	"time"
+
+	"net"
+
+	"github.com/dedis/cothority/log"
+)
+
+type TLSCertPEM []byte
+type TLSKeyPEM []byte
+
+// NewTLSCert returns a x509-certificate valid for all CommonNames.
+func NewTLSCert(serial *big.Int, country, org, orgUnit string,
+	validYear int, subjectKeyID []byte,
+	ips []net.IP) *x509.Certificate {
+	return &x509.Certificate{
+		SerialNumber: serial,
+		Subject: pkix.Name{
+			Country:            []string{country},
+			Organization:       []string{org},
+			OrganizationalUnit: []string{orgUnit},
+			CommonName:         "*",
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(validYear, 0, 0),
+		SubjectKeyId:          subjectKeyID,
+		BasicConstraintsValid: true,
+		IsCA:        true,
+		IPAddresses: ips,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage: x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign |
+			x509.KeyUsageKeyAgreement,
+	}
+}
+
+// NewTLSKC returns a TLSKC-structure with a PEM-encoded certificate and key.
+func NewCertKey(ca *x509.Certificate, keyLen int) (cert TLSCertPEM, key TLSKeyPEM, err error) {
+	if keyLen < 2048 {
+		log.Warn("Small key-length:", keyLen)
+	}
+	priv, _ := rsa.GenerateKey(rand.Reader, keyLen)
```

Comment:
Addition: Goland supports the ECDSA key ([link](https://golang.org/pkg/crypto/x509/#CreateCertificate)):
> All keys types that are implemented via crypto.Signer are supported (This includes *rsa.PublicKey and *ecdsa.PublicKey.)